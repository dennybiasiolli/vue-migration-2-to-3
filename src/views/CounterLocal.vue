<template>
  <div>
    <p>Count is: {{ count }}</p>
    <p>Number is {{ isEven ? 'even' : 'odd' }}</p>
    <button @click="() => increment()">Increment</button>
  </div>
</template>

<script>
export default {
  data() {
    return { count: 0 }
  },
  computed: {
    isEven() {
      return this.count % 2 === 0
    },
  },
  methods: {
    increment() {
      this.count++
    },
  },
}
</script>
