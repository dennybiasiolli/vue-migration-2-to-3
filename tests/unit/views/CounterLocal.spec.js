import { shallowMount } from '@vue/test-utils'
import CounterLocal from '@/views/CounterLocal.vue'

describe('CounterLocal.vue', () => {
  it('renders as expected', () => {
    const wrapper = shallowMount(CounterLocal)
    expect(wrapper).toMatchSnapshot()
  })

  it('should call increment after button click', () => {
    const wrapper = shallowMount(CounterLocal)
    // FIXME: spying on `wrapper.vm._setupState` methods is a temporary workaround
    const spyIncrement = jest.spyOn(wrapper.vm._setupState, 'increment')
    wrapper.find('button').trigger('click')
    expect(spyIncrement).toHaveBeenCalled()
  })
})
