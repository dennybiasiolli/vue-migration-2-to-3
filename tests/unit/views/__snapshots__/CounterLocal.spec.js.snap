// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterLocal.vue renders as expected 1`] = `
<div>
  <p>Count is: 0</p>
  <p>Number is even</p> <button>Increment</button>
</div>
`;
